### DISCUSSION TITLE
How to sweep a profile along an edge

### DISCUSSION THREAD

**Discussion by JanFilipec** [nortikin/sverchok#4579] [Q&A]
## Problem statement

How do I sweep a 2D profile along a planar mesh edge with sharp corners and get a clean result?

## Steps to reproduce

I tried using "bevel curve node", but it doesn't model the corner properly.

## Expected result
![grafik](https://user-images.githubusercontent.com/67052298/179311474-b5f61385-71e9-4a13-bb04-c98bdf490917.png)

## Actual result

![grafik](https://user-images.githubusercontent.com/67052298/179310444-e4e451cd-e3b9-4330-86d4-3cba3dd02ceb.png)

**Comment by zeffii** [comment]
i see what you mean..
![image](https://user-images.githubusercontent.com/619340/179391458-b2640dcf-f1b2-4864-9cc7-faf2257a5a4d.png)
[bevel_curve_test_2022_07_17_09_03.zip](https://github.com/nortikin/sverchok/files/9127312/bevel_curve_test_2022_07_17_09_03.zip)

in the 90 angle the bevel shape would need to sheer/transform to retain the shape properties.

**Comment by Durman** [comment]
@JanFilipec I glad that you asked. It's really simply to do in Sverchok. Node tree just as simple as that is quire enough.

![image](https://user-images.githubusercontent.com/28003269/179694698-10be6cd1-ff18-4881-b39b-b49954a8077b.png)

![profile polyline](https://user-images.githubusercontent.com/28003269/179695047-5cd96aa4-77a0-49ee-957d-490f29ffd6af.gif)

[SV Profile Polyline.zip](https://github.com/nortikin/sverchok/files/9138719/SV.Profile.Polyline.zip)

To speak more seriously it seems worth converting these experiments #2242 into a node to simplify the problem.
